Libplanet/Store/DefaultStore.cs

```diff
@@ -524,20 +528,36 @@ HashDigest<SHA256> blockHash
                 return cached;
             }
 
-            LiteFileInfo file =
-                _db.FileStorage.FindById(BlockStateFileId(blockHash));
-            if (file is null)
+            UPath path = StatePath(blockHash);
+            if (!_states.FileExists(path))
             {
                 return null;
             }
 
-            using (var stream = new MemoryStream())
+            using (var buffer = new MemoryStream())
+            using (Stream file = _states.OpenFile(path, System.IO.FileMode.Open, FileAccess.Read))
             {
-                DownloadFile(file, stream);
-                stream.Seek(0, SeekOrigin.Begin);
+                if (_compress)
+                {
+                    using (var deflate = new DeflateStream(file, CompressionMode.Decompress))
+                    {
+                        deflate.CopyTo(buffer);
+                    }
+                }
+                else
+                {
+                    file.CopyTo(buffer);
+                }
 
-                var deserialized = (Bencodex.Types.Dictionary)_codec.Decode(stream);
-                ImmutableDictionary<Address, IValue> states = deserialized.ToImmutableDictionary(
+                IValue value = new Codec().Decode(buffer.ToArray());
```

Comment:
`Bencodex.Codec` has an overload `Decode(Stream)` too.  You could reuse the `buffer` object by `buffer.Seek(0)` without allocating a new array.